### PR TITLE
feat(tools):Cache logs player

### PR DESCRIPTION
**Pull request by ashotland**
Signed-off-by: ashotland <ari@dragonflydb.io>

### CODE REVIEW

File: tools/cache_logs_player.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    async def work(self) -> None:
+        self.working = True
+        batch = []
+        while self.working or len(batch) > 0 or not self.queue.empty() :
+            try:
+                cmd = await asyncio.wait_for(self.queue.get(), timeout=1.0)
+                batch.append(cmd)
+                if len(batch) >= self.BATCH_SIZE:
+                    await self.execute(batch)
+                    batch.clear()
```
Comment (dranikpg) [inline tools/cache_logs_player.py:80]: If you group the values on the producer side instead and send them over the channel in batches you would reduce the pressure on the asyncio scheduler

But I suspect its not a bottleneck here, besides managing the timeout to send half filled batches from the producer would be a little inconvenient

**Review by dranikpg — APPROVED** [review]

File: tools/cache_logs_player.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        elif operation == 'set':
+            cmd.args = ["SET", key, synthetic_value]
+        elif operation == 'add':
+            cmd.args = ["SET", key, synthetic_value]
+        elif operation == 'replace':
+            cmd.args = ["SET", key, synthetic_value]
+        elif operation == 'cas':
+            cmd.args = ["SET", key, synthetic_value]
+        elif operation == 'append':
+            cmd.args = ["SET", key, synthetic_value]
```
Comment (romange) [inline tools/cache_logs_player.py:47]: @ashotland  redis has append I think

File: tools/cache_logs_player.py (hunk 1 above)
Comment (ashotland) [reply tools/cache_logs_player.py:80]: @dranikpg thanks, good feedback.

File: tools/cache_logs_player.py (hunk 2 above)
Comment (ashotland) [reply tools/cache_logs_player.py:47]: Correct, will fix.

**Pull request merged**